files/templates/mbta/account/index.ftl

```diff
@@ -43,9 +43,11 @@
                 isEditUserNameAllowed : ${realm.editUsernameAllowed?c},
                 isInternationalizationEnabled : ${realm.isInternationalizationEnabled()?c},
                 isLinkedAccountsEnabled : ${realm.identityFederationEnabled?c},
-                isEventsEnabled : ${isEventsEnabled?c},
                 isMyResourcesEnabled : ${(realm.userManagedAccessAllowed && isAuthorizationEnabled)?c},
-                isTotpConfigured : ${isTotpConfigured?c},
+                /* Per docs, these two were deprecated for a while, then removed in v22.0.0
```

Comment:
question: can you say more about the error you're seeing? I'd have expected to see this in production, but as far as I can test, account management works there.

Reply:
```
keycloak-1  | 2024-05-20 18:21:44,070 SEVERE [freemarker.runtime] (executor-thread-2) Error executing FreeMarker template: freemarker.core.InvalidReferenceException: The following has evaluated to null or missing:
keycloak-1  | ==> isEventsEnabled  [in template "index.ftl" at line 49, column 40]
keycloak-1  |
keycloak-1  | ----
keycloak-1  | Tip: If the failing expression is known to legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
keycloak-1  | ----
keycloak-1  |
keycloak-1  | ----
keycloak-1  | FTL stack trace ("~" means nesting-related):
keycloak-1  | 	- Failed at: ${isEventsEnabled?c}  [in template "index.ftl" at line 49, column 38]
keycloak-1  | ----
keycloak-1  | 	at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:134)
keycloak-1  | 	at freemarker.core.BuiltInsForMultipleTypes$cBI.formatNull(BuiltInsForMultipleTypes.java:61)
keycloak-1  | 	at freemarker.core.BuiltInsForMultipleTypes$AbstractCLikeBI._eval(BuiltInsForMultipleTypes.java:92)
keycloak-1  | 	at freemarker.core.Expression.eval(Expression.java:101)
keycloak-1  | 	at freemarker.core.DollarVariable.calculateInterpolatedStringOrMarkup(DollarVariable.java:104)
keycloak-1  | 	at freemarker.core.DollarVariable.accept(DollarVariable.java:63)
keycloak-1  | 	at freemarker.core.Environment.visit(Environment.java:335)
keycloak-1  | 	at freemarker.core.Environment.visit(Environment.java:341)
keycloak-1  | 	at freemarker.core.Environment.process(Environment.java:314)
keycloak-1  | 	at freemarker.template.Template.process(Template.java:383)
keycloak-1  | 	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:49)
keycloak-1  | 	at org.keycloak.services.resources.account.AccountConsole.getMainPage(AccountConsole.java:161)
keycloak-1  | 	at org.keycloak.services.resources.account.AccountConsole$quarkusrestinvoker$getMainPage_d508fd32a638b899af4d52f90082abf4f245fdb3.invoke(Unknown Source)
keycloak-1  | 	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
keycloak-1  | 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
keycloak-1  | 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
keycloak-1  | 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
keycloak-1  | 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
keycloak-1  | 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
keycloak-1  | 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
keycloak-1  | 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
keycloak-1  | 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
keycloak-1  | 	at java.base/java.lang.Thread.run(Thread.java:840)
keycloak-1  |
keycloak-1  | 2024-05-20 18:21:44,074 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-2) Uncaught server error: org.keycloak.theme.FreeMarkerException: Failed to process template index.ftl
keycloak-1  | 	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:52)
keycloak-1  | 	at org.keycloak.services.resources.account.AccountConsole.getMainPage(AccountConsole.java:161)
keycloak-1  | 	at org.keycloak.services.resources.account.AccountConsole$quarkusrestinvoker$getMainPage_d508fd32a638b899af4d52f90082abf4f245fdb3.invoke(Unknown Source)
keycloak-1  | 	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
keycloak-1  | 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
keycloak-1  | 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
keycloak-1  | 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
keycloak-1  | 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
keycloak-1  | 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
keycloak-1  | 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
keycloak-1  | 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
keycloak-1  | 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
keycloak-1  | 	at java.base/java.lang.Thread.run(Thread.java:840)
keycloak-1  | Caused by: freemarker.core.InvalidReferenceException: The following has evaluated to null or missing:
keycloak-1  | ==> isEventsEnabled  [in template "index.ftl" at line 49, column 40]
keycloak-1  |
keycloak-1  | ----
keycloak-1  | Tip: If the failing expression is known to legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
keycloak-1  | ----
keycloak-1  |
keycloak-1  | ----
keycloak-1  | FTL stack trace ("~" means nesting-related):
keycloak-1  | 	- Failed at: ${isEventsEnabled?c}  [in template "index.ftl" at line 49, column 38]
keycloak-1  | ----
keycloak-1  | 	at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:134)
keycloak-1  | 	at freemarker.core.BuiltInsForMultipleTypes$cBI.formatNull(BuiltInsForMultipleTypes.java:61)
keycloak-1  | 	at freemarker.core.BuiltInsForMultipleTypes$AbstractCLikeBI._eval(BuiltInsForMultipleTypes.java:92)
keycloak-1  | 	at freemarker.core.Expression.eval(Expression.java:101)
keycloak-1  | 	at freemarker.core.DollarVariable.calculateInterpolatedStringOrMarkup(DollarVariable.java:104)
keycloak-1  | 	at freemarker.core.DollarVariable.accept(DollarVariable.java:63)
keycloak-1  | 	at freemarker.core.Environment.visit(Environment.java:335)
keycloak-1  | 	at freemarker.core.Environment.visit(Environment.java:341)
keycloak-1  | 	at freemarker.core.Environment.process(Environment.java:314)
keycloak-1  | 	at freemarker.template.Template.process(Template.java:383)
keycloak-1  | 	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:49)
keycloak-1  | 	... 12 more
```

Reply:
[Changelog record of the removal of these variables is here.](https://www.keycloak.org/docs/latest/release_notes/#account-console-template-variables-removed)

Reply:
@integsoft-company can you say more about this? I'm a little confused about how this template is working in production if the variables were removed.

Reply:
@jurakp

Reply:
Will check with my colleague tomorrow and let you know.

Reply:
@paulswartz We will create a new PR with the fixed template (the current is obsolete and not used, I mean these part causing issues in this PR).
First we will deploy it to the sandbox. Will let you know once done.

Reply:
Now it is in the sandbox and a new PR is here: https://github.com/mbta/keycloak-deploy/pull/30
This PR resolves the new templates, which were missing during KC upgrade. So after merging you can do whatever needed. So I would recommend to close PR #29 and #28 and replay commits on the top of this PR.

Reply:
I've rebased this branch on current main.